pkg/query/storeset.go

```diff
@@ -397,3 +408,17 @@ func (s *StoreSet) Close() {
 		st.close()
 	}
 }
+
+func (s *StoreSet) cleanUpStoreStatuses() {
+	s.storesStatusesMtx.RLock()
```

Comment:
I feel like we need rather `Lock` as we not only read but also "write" statuses -> particularly delete.

Reply:
Right ! Still learning about those